### PR TITLE
Add/update Projects and Datasets methods: remove_permanently, get_list_all

### PR DESCRIPTION
**Pull request by GoldenAnpu**
Changes: 
 - project_api 
    - add team_id in ProjectInfo
    - add def remove_permanently
    - add def get_list_all
    - update get_archivation_list to work via get_list_all for backwards compatibility
 - dataset_api
    - add team_id and workspace_id in DatasetInfo
    - update type for arg 'page' in def get_list_all and its docstring
    - update return for def get_list_all to return entities as DatasetInfo
 - module_api
    - add ApiField - PRESERVE_PROJECT_CARD

### CODE REVIEW

File: supervisely/api/dataset_api.py [hunk 1]
```diff
@@ -565,9 +572,9 @@ def get_list_all(
 
         :param page: Page number, used to retrieve the following items if the number of them found is more than per_page.
```
Comment (mkolomeychenko) [inline supervisely/api/dataset_api.py:573]: None

File: supervisely/api/project_api.py [hunk 2]
```diff
@@ -1274,23 +1279,45 @@ def get_archivation_list(
...
+                "value": date,
+            }
+            filters.append(filer_from)
         if to_day is not None:
-            request_body[ApiField.TO] = to_day
-        if skip_exported is not None:
-            request_body[ApiField.SKIP_EXPORTED] = skip_exported
+            date = (datetime.utcnow() - timedelta(days=to_day)).strftime("%Y-%m-%dT%H:%M:%SZ")
+            filer_to = filer_from = {
+                "field": "updatedAt",
```
Comment (mkolomeychenko) [inline supervisely/api/project_api.py:1296]: ApiField.UPDATED_AT

File: supervisely/api/project_api.py [hunk 3]
```diff
@@ -1274,23 +1279,45 @@ def get_archivation_list(
...
         """
-        request_body = {}
+        kwargs = {}
+
+        filters = []
         if from_day is not None:
-            request_body[ApiField.FROM] = from_day
+            date = (datetime.utcnow() - timedelta(days=from_day)).strftime("%Y-%m-%dT%H:%M:%SZ")
+            filer_from = {
+                "field": "updatedAt",
```
Comment (mkolomeychenko) [inline supervisely/api/project_api.py:1288]: ?

File: supervisely/api/project_api.py [hunk 4]
```diff
@@ -1274,23 +1279,45 @@ def get_archivation_list(
...
+            filters.append(filer_to)
+        if len(filters) != 0:
+            kwargs["filters"] = filters
+
+        if skip_exported is None:
+            kwargs["skip_exported"] = skip_exported
+
         if sort is not None:
-            request_body[ApiField.SORT] = sort
+            kwargs["sort"] = sort
```
Comment (mkolomeychenko) [inline supervisely/api/project_api.py:1308]: ?

File: supervisely/api/project_api.py [hunk 5]
```diff
@@ -1438,3 +1465,189 @@ def set_multiview_settings(self, project_id: int) -> None:
             tag_name=_MULTIVIEW_TAG_NAME,
             sync=False,
         )
+
+    def remove_permanently(self, ids: Union[int, List]) -> dict:
+        """
+        Delete permanently projects with given IDs from the Supervisely server.
```
Comment (mkolomeychenko) [inline supervisely/api/project_api.py:1471]: ?

Comment (mkolomeychenko) [reply supervisely/api/project_api.py:1471]: docs

**Review by mkolomeychenko — CHANGES_REQUESTED** [review]

File: supervisely/api/project_api.py (hunk 4 above)
Comment (GoldenAnpu) [reply supervisely/api/project_api.py:1308]: this kwargs for def get_list_all

**Pull request merged**